vmm/src/device_manager.rs

```diff
@@ -403,6 +422,52 @@ pub fn get_win_size() -> (u16, u16) {
     (ws.cols, ws.rows)
 }
 
+const TIOCSPTLCK: libc::c_int = 0x4004_5431;
+const TIOCGTPEER: libc::c_int = 0x5441;
+
+pub fn create_pty() -> io::Result<(File, File, PathBuf)> {
+    let main = match OpenOptions::new()
+        .read(true)
+        .write(true)
+        .custom_flags(libc::O_NOCTTY)
+        .open("/dev/pts/ptmx")
+    {
+        Ok(f) => f,
+        _ => OpenOptions::new()
+            .read(true)
+            .write(true)
+            .custom_flags(libc::O_NOCTTY)
+            .open("/dev/ptmx")?,
+    };
+    let mut unlock: libc::c_ulong = 0;
+    unsafe {
+        libc::ioctl(
+            main.as_raw_fd(),
+            TIOCSPTLCK.try_into().unwrap(),
+            &mut unlock,
+        )
+    };
```

Comment:
You should check if `TIOCSPTLCK` has been correctly applied by checking the returned value from `ioctl()`.

Reply:
See my comment [above](https://github.com/cloud-hypervisor/cloud-hypervisor/pull/2168#discussion_r568040642) for this.

Reply:
Thanks for the explanation, sounds good to me.